ir/pattern/optional.rs

```diff
@@ -40,6 +41,24 @@ impl Optional {
     pub fn conjunction_mut(&mut self) -> &mut Conjunction {
         &mut self.conjunction
     }
+
+    pub(crate) fn variable_dependency(
+        &self,
+        block_context: &BlockContext,
+    ) -> HashMap<Variable, VariableDependency<'_>> {
+        self.conjunction
+            .variable_dependency(block_context)
+            .into_iter()
+            .map(|(var, mut mode)| {
+                // VariableDependency::Producing means "producing in all branches".
+                // A try {} block never produces.
```

Comment:
Is this a fact?

Reply:
Yep. An optional variablet is not `Produced` for the purposes of resolving `Require`ments such as in an expression or a function argument.